wandb/sdk/service/server_sock.py

```diff
@@ -0,0 +1,257 @@
+import queue
+import socket
+import threading
+import time
+from typing import Any, Callable, Dict, Optional
+from typing import TYPE_CHECKING
+
+from wandb.proto import wandb_server_pb2 as spb
+
+from .streams import _dict_from_pbmap
+from .streams import StreamMux
+from ..lib.sock_client import SockClient
+
+
+if TYPE_CHECKING:
+    from threading import Event
+    from ..interface.interface_relay import InterfaceRelay
+
+
+class ClientDict:
```

Comment:
what do you think about using dunder methods (i.e. `__setitem__`, `__getitem__` and `__delitem__`)?

Reply:
If this was a more critical datastructure shared between more parts of the code, i agree this would be more readable to give this a dict interface.
i think as it is right now it is ok, more verbose than needed, but clear what is going on.